packages/backend/src/services/spsp.ts

```diff
@@ -0,0 +1,91 @@
+import Axios from 'axios'
+import base64url from 'base64url'
+import * as uuid from 'uuid'
+import { Middleware } from 'koa'
+import { StreamServer } from '@interledger/stream-receiver'
+import { AppContainer, AppContext } from '../app'
+import { User } from '../models'
+
+export async function makeSPSPHandler(
+  container: AppContainer
+): Promise<Middleware<unknown, AppContext>> {
+  const config = await container.use('config')
+  const axios = Axios.create({
+    baseURL: config.accountsUrl,
+    timeout: 10_000,
+    validateStatus: (status) => status === 200 || status === 404
+  })
+  const server = new StreamServer({
+    serverSecret: config.streamSecret,
+    serverAddress: config.ilpAddress
+  })
+
+  return async function handleSPSP(ctx: AppContext): Promise<void> {
+    if (!uuid.validate(ctx.params.id) || uuid.version(ctx.params.id) !== 4) {
```

Comment:
Should the version number be made a constant somewhere?  Will this `4` need to be updated at some point?

Reply:
We always generate uuid version-4 (the "version" is really a different algorithm). But I'll remove that check. The validation here is only to ensure that the UUID is valid so that knex doesn't throw a validation error.

Reply:
I'll defer to your judgement on that.  I'm not sure of the proper way to validate the uuid.  My only concern is having version numbers for things hidden in code versus a config file or a constants file.